docs/package-management.md

```diff
@@ -0,0 +1,137 @@
+
+The Spack documentation provides a [tutorial on
+environments](https://spack-tutorial.readthedocs.io/en/latest/tutorial_environments.html),
+which is useful for a more comprehensive understanding of how these environments
+function. In this documentation, we'll take a more focused approach, dealing
+with common setup and installation tasks for preparing an environment for a
+course.
+
+You can only set up environments in environments that you can modify. In HUIT
+Open OnDemand, that means that you can only set up environments in your own
+personal Spack installation in your home folder, or in a shared course folder.
+In both cases, you should have the global spack installation set as an upstream,
+whether by the course shared folder's pre-made spack installation script, or by
+following our directions on setting up your own Spack installation.
+
+For the rest of this document, we'll assume that you've already sourced the
+`setup-env.sh` script from the installation where you want the environment to
+be.
+
+### Creating & Activating
+
+For an environment to be used, it must first be created. This is pretty
+straightforward, with the command
+
+```bash
+spack env create environmentName
+```
+
+Replace "environmentName" with whatever you want the environment to be called.
+Then the environment can be activated with
+
+```bash
+spack env activate environmentName
+```
+
+### Adding Packages
+
+To get started, you can add one or more packages. You do this with a command like
+
+```bash
+spack add packageSpec
+```
+
+You can install a package simply by name, but if you have specific version or
+compiler requirements, you can specify those with [Spack's spec
+syntax](https://spack.readthedocs.io/en/latest/basic_usage.html#sec-specs).
+
+Adding a package won't install it, but it will associate that package with the
+environment itself.
+
+### Installing Packages
+
+The syntax to start an installation is simple, but where to run the installation
+is less simple. It's best to run these installations in a Slurm job, since some
+installations will take a long time to run. Therefore, we recommend adapting
+this bash script for your Spack installation, and using it via `sbatch`. This is
+the script that we use to manage the global installation, so just change the
+line with the Spack activation to match your installation path.
+
+```bash
+#!/bin/bash
+if [ -x "${1}" ]
+  then
+    echo "Environment name is required, e.g. `spackEnvironment.sh environmentName`"
+    exit 1
+fi
+
+# Change the line below to activate your own Spack environment
+. /shared/spack/share/spack/setup-env.sh # CHANGEME
+ENVIRONMENT=$1
+echo "Activating environment $ENVIRONMENT..."
+spack env activate $ENVIRONMENT
+echo "Installing packages for $ENVIRONMENT..."
+spack install
+echo "Done!"
+```
+
+From wherever you place that script, assuming you name it `spackEnvironment.sh`,
+you can run it as a Slurm job with
+
+```bash
+sbatch -c 4 spackEnvironment.sh environmentName
```

Comment:
We should also add: If you want to list the Slurm jobs that are currently running you can run the following commands `squeue -u <username>` replacing the <username> with your username.
Note: username can be found on the landing page -> upper right corner

<img width="1908" alt="Screenshot 2024-08-22 at 9 42 05 AM" src="https://github.com/user-attachments/assets/7201d8a9-b6a4-4c50-9fac-a2cbc7af8be5">